src/Emulator/Tabs/ConfigureTab/ConfigureTab.js

```diff
@@ -39,29 +39,65 @@ const hostConfig = [
   },
 ];
 
+const getNextNumber = (s) => Number(s.slice(1)) + 1;
+const getNextDeviceName = (device) => {
```

Comment:
Just confirming my understanding... if we have:

`s1`, `s2`, `s3`
[add]
`s1`, `s2`, `s3`, `s4`
[delete]
`s1`, `s2`, `s3`
[add]
`s1`, `s2`, `s3`, `s4`

We want `s4` not `s5` there. I think that's elegant just want to make sure.

What happens if a user deletes all devices in a queue?

Reply:
Yeah so as I said in the meeting, I did not think about delete interactions when I wrote this code. When the user tries to hit delete but there isn't anything to delete, we can either have some kind of error message or just disable the button and make it visually clear that clicking it won't do anything.

As for the other issue, it's definitely a bit more complex. If we already have 10 devices, like s1, s2, ..., s10, and we delete s2 for example, we would need to decide if the new switch should be s2 or s11. My immediate thought is that it should be s11 because making it s2 seems weird, and because the scroll to the bottom of the device container would break.

With that being said, I think this is worth discussing more in-depth with the entire team.

Reply:
Sounds good - let's take care of the delete when there's nothing case since the code should have sufficient safeties for future possibilities (and we are sure we are doing this one), but we can have the other case (`s2` or `s11`) be handled in subsequent PRs.